packages/repo-metadata-lint/src/bin/commands/scan-repo.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import yargs = require('yargs');
+import {octokitFrom} from '../../utils/octokit-util';
+import {GetRepoFiles} from '../../get-repo-files';
+import {IssueOpener} from '../../issue-opener';
+import {Validate, ValidationResult} from '../../validate';
+
+interface Args {
+  'pem-path': string;
+  'app-id': number;
+  installation: number;
+  owner: string;
+  repo: string;
+  'commit-cache-bucket': string;
+}
+
+export const scanRepo: yargs.CommandModule<{}, Args> = {
+  command: 'scan-repo',
+  describe:
+    'scans a repository opening issues for corrupt .repo-metadata.json files',
+  builder(yargs) {
+    return yargs
+      .option('pem-path', {
+        describe: 'provide path to private key for requesting JWT',
```

Comment:
This is a little weird - in the CIs I've built I just ask for a path to a personal access token to be set in `GITHUB_TOKEN`.

Reply:
I pulled this approach over from `owl-bot`, what I like about it is I can test exactly the credentials that the automation is using, which makes it easy to debug a missing permission on the key, or a repository I've forgotten to install the automation on -- I find it's quite useful once the automation is actually live and I'm investigating a bug report.